activesupport/lib/active_support/callbacks.rb

```diff
@@ -642,11 +643,21 @@ def initialize_copy(other)
           @mutex     = Mutex.new
         end
 
-        def compile
-          @callbacks || @mutex.synchronize do
-            final_sequence = CallbackSequence.new
-            @callbacks ||= @chain.reverse.inject(final_sequence) do |callback_sequence, callback|
-              callback.apply callback_sequence
+        def compile(types)
+          if types.nil?
+            @callbacks || @mutex.synchronize do
+              final_sequence = CallbackSequence.new
+              @callbacks ||= @chain.reverse.inject(final_sequence) do |callback_sequence, callback|
+                callback.apply callback_sequence
+              end
+            end
+          else
+            types = Array(types)
```

Comment:
Since this is low level API, I prefer to always receive an array, so we avoid the allocation here